### DISCUSSION TITLE
Controller component does not pass field value type to render prop

### DISCUSSION THREAD

**Discussion by foxylion** [react-hook-form/react-hook-form#6334] [Q&A] [answered]
### Version Number

7.4.1

### Codesandbox/Expo snack

https://codesandbox.io/s/adoring-hodgkin-r445l?file=/src/App.tsx

### Steps to reproduce

1. Open the linked codesandbox
2. See code contains typescript error

### Expected behaviour

The type of the `field.value` property should be `number | undefined` and not `string | number | undefined`.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Relevant log output

```shell
(JSX attribute) value: number | undefined
Type 'string | number | undefined' is not assignable to type 'number | undefined'.
  Type 'string' is not assignable to type 'number | undefined'.ts(2322)
App.tsx(13, 33): The expected type comes from property 'value' which is declared here on type 'IntrinsicAttributes & { value: number | undefined; } & { children?: ReactNode; }'
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
hope the following helps:

https://codesandbox.io/s/admiring-glitter-xu1jc?file=/src/App.tsx

**Comment by foxylion** [comment, marked as answer]
Hi @bluebill1049,
thank you for your super fast reply. :-)

I see you just removed the generics type definition. This basically just removed any type-safety from the component. That said, yes it does now compile, but here is no more safety by Typescript.

One example is that I can now use any value for the `name` prop and it is not checked by the compiler if this is a valid key from the `TFieldValues` type. Here is an (obviously broken) example: https://codesandbox.io/s/exciting-water-bzjnc

Any ideas how correct typing could be preseved without the need to "disable" the type checking?

**Reply by bluebill1049** [reply]
i see.

```
<Controller<FormValues, "fooBar">
  name="fooBar"
  render={({ field }) => <MyCustomInput value={field.value} />}
/>
```

the FieldPathValue will need the input field name, so template string literal can look up the correct type.

**Reply by foxylion** [reply]
Awesome, this seems to do the trick. Thank you a lot!
